scripts/open-coredump.sh

```diff
@@ -0,0 +1,291 @@
+    scylla-enterprise.git) will be cloned from https://github.com/scylladb.
+    The script assumes you have the appropriate access rights set up.
+    Can be also provided via env variable SCYLLA_REPO_PATH.
+    If both are provided command line has precedence.
+
+--scylla-gdb-py-source repo|package|none
+    Where to get the scylla-gdb.py script from:
+    * repo - download the latest one from the appropriate main branch;
+    * package - use the one that comes with the relocatable package;
+    * none - don't try to obtain scylla-gdb.py;
+    Defaults to repo.
+    Unlike other artifacts, scylla-gdb.py is placed in the same directory
+    where the coredump is located. If one is already present, it is not
+    overwritten.
+    Can be also provided via env variable SCYLLA_GDB_PY_SOURCE.
+    If both are provided command line has precedence.
+EOF
+}
+
+function log {
+    if [[ $VERBOSE_LEVEL -ge 1 ]]
+    then
+        echo $@
+    fi
+}
+
+for required in eu-unstrip jq curl git; do
+    if ! type $required >& /dev/null; then
+        echo "error: missing required program $required, please install first"
+        exit 1
+    fi
+done
+
+VERBOSE_LEVEL=1
+SCYLLA_S3_RELOC_SERVER_URL="${SCYLLA_S3_RELOC_SERVER_URL:-http://backtrace.scylladb.com}"
+SCYLLA_REPO_PATH="${SCYLLA_REPO_PATH}"
+SCYLLA_GDB_PY_SOURCE="${SCYLLA_GDB_PY_SOURCE:-repo}"
+ARTIFACT_DIR=${ARTIFACT_DIR:-open-coredump.sh.dir}
+
+while [[ $# -gt 1 ]]
+do
+    case $1 in
+        "--help"|"-h")
+            print_usage
+            exit 0
+            ;;
+        "--progress"|"-n")
+            VERBOSE_LEVEL=2
+            shift 1
+            ;;
+        "--quiet"|"-q")
+            VERBOSE_LEVEL=0
+            shift 1
+            ;;
+        "--artifact-dir"|"-d")
+            ARTIFACT_DIR=$2
+            shift 2
+            ;;
+        "--scylla-s3-reloc-server-url")
+            SCYLLA_S3_RELOC_SERVER_URL=$2
+            shift 2
+            ;;
+        "--scylla-repo-path")
+            SCYLLA_REPO_PATH=$2
+            shift 2
+            ;;
+        "--scylla-gdb-py-source")
+            SCYLLA_GDB_PY_SOURCE=$2
+            shift 2
+            ;;
+        *)
+            echo "unrecognized option: $1, see $0 -h for usage"
+            exit 1
+            ;;
+    esac
+done
+
+COREFILE=$1
+shift 1
+if ! [[ -f $COREFILE ]]
+then
+    echo "error: ${COREFILE} is not a valid path to a core file"
+    exit 1
+fi
+COREDIR=$(dirname ${COREFILE})
+
+ARTIFACT_DIR=$(realpath ${ARTIFACT_DIR})
+mkdir -p $ARTIFACT_DIR
+
+if [[ -z "${SCYLLA_REPO_PATH}" ]]
+then
+    SCYLLA_REPO_PATH="${ARTIFACT_DIR}/scylla.repo"
+fi
+
+SCYLLA_REPO_PATH=$(realpath ${SCYLLA_REPO_PATH})
+
+if [[ ${VERBOSE_LEVEL} -lt 2 ]]
+then
+    CURL_QUIET_FLAG="-s"
+    GIT_QUIET_FLAG="-q"
+else
+    CURL_QUIET_FLAG=""
+    GIT_QUIET_FLAG=""
+fi
+
+if ! [[ "${SCYLLA_GDB_PY_SOURCE}" == "repo" ]] && ! [[ "${SCYLLA_GDB_PY_SOURCE}" == "package" ]] && ! [[ "${SCYLLA_GDB_PY_SOURCE}" == "none" ]]
```

Comment:
nit: I think you can use the switch like `case` notation more elegantly for multiple value choices of a particular variable, like:
```bash
case "${SCYLLA_GDB_PY_SOURCE}" in
(repo|package|none)
    ;;
*)
    echo "error: invalid value for option SCYLLA_GDB_PY_SOURCE, has to be one of repo, package or none, got: ${SCYLLA_GDB_PY_SOURCE}"
    exit 1;;
esac
```

Reply:
Makes sense, indeed it looks better.

Reply:
Done in v4.